File: Android/app/src/main/java/com/denarii/android/user/WalletDetails.java
package com.denarii.android.user;

import java.io.Serializable;

public class WalletDetails implements Serializable
{
    private String walletName;

    private String walletPassword;

    private String seed;

    private double balance;

    private String walletAddress;

    public String getWalletName(){return walletName;}

    public String getWalletPassword(){return walletPassword;}

    public String getSeed(){return seed;}

    public double getBalance(){return balance;}

    public String getWalletAddress(){return walletAddress;}

    public void setWalletName(String newName){walletName = newName;}

    public void setWalletPassword(String newPassword){walletPassword = newPassword;}

    public void setSeed(String newSeed){seed = newSeed;}

    public void setBalance(double newBalance){balance = newBalance;}

    public void setWalletAddress(String newAddress){walletAddress = newAddress;}

}

File: Ios/denarii/denarii/user/WalletDetails.swift
//
//  WalletDetails.swift
//  denarii
//
//  Created by Andrew Katson on 9/30/23.
//

import Foundation

class WalletDetails {
    var walletName : String = ""

    private(set) var walletPassword : String = ""

    var seed : String = ""

    var balance : Double = 0.0

    var walletAddress : String = ""
    
    init(walletName: String, walletPassword: String, seed: String, balance: Double, walletAddress: String) {
        self.walletName = walletName
        self.walletPassword = walletPassword
        self.seed = seed
        self.balance = balance
        self.walletAddress = walletAddress
    }
    
    init() {
        self.walletName = ""
        self.walletPassword = ""
        self.seed = ""
        self.balance = 0.0
        self.walletAddress = ""
    }
}
